s/scipy/build_info.json

```diff
@@ -1,10 +1,10 @@
 {
-   "maintainer": "Salil4IBM",
+   "maintainer": "puneetsharma21",
    "package_name": "scipy",
    "github_url": "https://github.com/scipy/scipy",
-   "version": "v1.14.1",
```

Comment:
Consider the latest version in build_info.json. We have build script for v1.14.1, So let's keep the same.